stacker/hooks/aws_lambda.py

```diff
@@ -61,8 +62,29 @@ def _zip_files(files, root):
 
     contents = zip_data.getvalue()
     zip_data.close()
+    content_hash = _calculate_hash(files, root)
 
-    return contents
+    return contents, content_hash
+
+
+def _calculate_hash(files, root):
+    """ Returns a hash of all of the given files at the given root.
+
+    Args:
+        files (list[str]): file names to include in the hash calculation,
+            relative to ``root``.
+        root (str): base directory to analyze files in.
+
+    Returns:
+        str: A hash of the hashes of the given files.
+    """
+    file_hashes = []
+    for fname in files:
+        f = os.path.join(root, fname)
+        with open(f) as fd:
+            file_hashes.append(hashlib.md5(fd.read()).hexdigest())
```

Comment:
a) It is probably a good idea to open the file with `rb` mode to avoid newline shenanigans.
b) This might not catch file renames if the files are still produced in the same order.

It might be necessary to sort files by their full path in the ZIP file, and prepend the path to the content before hashing.